### DISCUSSION TITLE
Collect ideas to get around SSL error on unrooted devices

### DISCUSSION THREAD

**Discussion by 573** [nix-community/nix-on-droid#341] [Ideas]
Probably related: #203.

Sometimes I get [SSL connect errors](https://discourse.nixos.org/t/ssl-connect-error-35/39316) with flakes on the device.

The `NIX_SSL_CERT_FILE` file hack does not always help, so I thought I would use `chronyd` for the purpose of setting correct time.

Creating this discussion mainly to gather intel on the mitigation of the SSL-Issue.

**Comment by 573** [comment]
Note: `chronyd` didn't work because I cannot root my phone:
> Fatal error : Not superuser 

Note: I will look into deploying config to my device via remote machine where I have full access, but that is only just a workaround.

**Comment by t184256** [comment]
On March 15, 2024 1:52:16 PM GMT+01:00, Daniel Kahlenberg ***@***.***> wrote:
>Probably related: #203.
>
>Sometimes I get [SSL connect errors](https://discourse.nixos.org/t/ssl-connect-error-35/39316) with flakes on the device.
>
>The `NIX_SSL_CERT_FILE` file hack does not always help, so I thought I would use `chronyd` for the purpose of setting correct time.
>
>Creating this discussion mainly to gather intel on the mitigation of the SSL-Issue.
>

Huh? Can you set the correct time in some simpler way?

**Comment by 573** [comment]
What I tried now is:
* (have a machine running Windows atm) installing `adb` there (`scoop install adb`)
* connecting the phone with usb debugging on (when connected the, selecting usb for picture transfer seemed to do it, see attached picture)
* on Windows machine run `adb shell settings put global ntp_server de.pool.ntp.org` and `adb reboot`

~Note: I forgot the `reboot` while testing rn thus the SSL error is still persisting.~

~Update: After `adb reboot` I retried in nix-on-droid and so far no SSL errors in the expected amount of time.~

Update 2: Must be something else still, the error is back.

picture;
![Screenshot_20240315-160234_Settings](https://github.com/nix-community/nix-on-droid/assets/123878/442b5547-7736-44ab-b547-7bada10d64ba)

**Comment by 573** [comment]
Indeed my phone's clock is continuing to lag:
```
-bash-5.2$ nix shell .#nixOnDroidConfigurations.sams9.pkgs.chrony --command chronyd -Q -t 3 'server de.pool.ntp.org iburst maxsamples 1'
2024-03-15T19:39:08Z chronyd version 4.4 starting (+CMDMON +NTP +REFCLOCK +RTC +PRIVDROP +SCFILTER +SIGND +ASYNCDNS +NTS +SECHASH +IPV6 -DEBUG)
2024-03-15T19:39:08Z Disabled control of system clock
2024-03-15T19:39:09Z System clock wrong by 3.193825 seconds (ignored)
2024-03-15T19:39:09Z chronyd exiting
```

**Comment by 573** [comment]
https://discourse.nixos.org/t/manually-set-date-and-time-on-nixos/13016 for simpler ways to set time.
